thetis/adr_model.py

```diff
@@ -0,0 +1,220 @@
+import firedrake as fd
+import networkx as nx
+import sympy
+from sympy.parsing.sympy_parser import \
+    convert_xor, \
+    parse_expr, \
+    standard_transformations
+from sympy.utilities.lambdify import lambdify
+
+
+class ADR_Model:
```

Comment:
This class was originally created for my MSc project. As such, it may need some modification to integrate better with Thetis. I have already adjusted the format of the docstrings, but the functionality is mostly untouched.